src/multimeter.js

```diff
@@ -167,46 +167,45 @@ module.exports = class Multimeter extends EventEmitter {
   }
 
   connect() {
-    this.console.log(`Connecting to ${(this.config.serverUrl ? ('(' + this.config.serverUrl + ')') : 'Screeps')} as ${this.config.email} ...`);
-    this.api.on('open', () => {
-      // Force a screen refresh to clear the screeps-api debug output
-      this.screen.alloc();
-      this.screen.render();
+    this.console.log(`Connecting to (${this.api.opts.url}) ...`);
 
-      this.api.subscribe('/console');
-      this.api.subscribe('/cpu');
-      this.api.subscribe('/code');
-    });
-
-    this.api.on('console', (msg) => {
-      const [, data] = msg;
+    this.api.socket.subscribe('console', (event) => {
+      const { data } = event;
       if (data.messages) {
         data.messages.log.forEach(l => this.console.addLines('log', l));
         data.messages.results.forEach(l => this.console.addLines('result', l));
       }
       if (data.error) this.console.addLines('error', data.error);
     });
 
-    this.api.on('message', (msg) => {
-      if (msg[0].slice(-4) == "/cpu") {
-        const cpu = msg[1].cpu, memory = msg[1].memory;
-        this.gauges.update(cpu, this.cpuLimit, memory, this.memoryLimit);
-      }
+    this.api.socket.subscribe('cpu', (event) => {
+        var { data } = event;
+        this.gauges.update(data.cpu, this.cpuLimit, data.memory, this.memoryLimit);
     });
 
-    this.api.on('code', (msg) => {
+    this.api.socket.subscribe('code', (msg) => {
       this.log('Code updated');
     });
 
-    this.api.on('close', () => {
+    /*
+    this.api.socket.on('message', (msg) => {
+        this.log(JSON.stringify(msg));
+    });
+    */
+
+    this.api.socket.on('connected', () => {
+      //this.log("Connected.");
```

Comment:
Do you not want to show this? I've found that since tick times can be several seconds, it's comforting to know that it's connected before I get the console messages from the first tick.

Reply:
I mainly left this here for the purposes of discussion. It's not in the original, which is why I commented it out. Just like your version, the MOTD isn't shown until the first connection.

Note that this connected event fires for reconnections as well as the first connection.